client/my-sites/media-library/list-item.tsx

```diff
@@ -0,0 +1,159 @@
+/**
+ * External dependencies
+ */
+import { isEqual, noop } from 'lodash';
+import React from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Spinner from 'components/spinner';
+import Gridicon from 'components/gridicon';
+import ListItemImage from './list-item-image';
+import ListItemVideo from './list-item-video';
+import ListItemAudio from './list-item-audio';
+import ListItemDocument from './list-item-document';
+import { getMimePrefix } from 'lib/media/utils';
+import EditorMediaModalGalleryHelp from 'post-editor/media-modal/gallery-help';
+
+/**
+ * Style dependencies
+ */
+import './list-item.scss';
+
+// TODO: move to lib/media/utils once it gets typed.
+interface MediaObject {
+	transient?: boolean;
+	file?: string;
+	[propName: string]: any;
+}
+type Media = string | MediaObject;
+// END TODO
+
+interface Props {
+	media?: Media;
+	scale: number;
+	maxImageWidth?: number;
+	thumbnailType?: string;
+	showGalleryHelp?: boolean;
+	selectedIndex?: number;
+	onToggle?: ( media: Media | undefined, shiftKey: boolean ) => any;
+	onEditItem?: any; // Unused. Appears to have been left here for compatibility reasons.
+	style?: React.CSSProperties;
+}
+
+type DivProps = Omit< React.ComponentProps< 'div' >, 'style' | 'onClick' >;
+
+export default class MediaLibraryListItem extends React.Component< Props & DivProps > {
+	static defaultProps = {
+		maxImageWidth: 450,
+		selectedIndex: -1,
+		onToggle: noop,
+		onEditItem: noop,
+	};
+
+	shouldComponentUpdate( nextProps: Props ) {
+		return ! (
+			nextProps.media === this.props.media &&
+			nextProps.scale === this.props.scale &&
+			nextProps.maxImageWidth === this.props.maxImageWidth &&
+			nextProps.thumbnailType === this.props.thumbnailType &&
+			nextProps.selectedIndex === this.props.selectedIndex &&
+			isEqual( nextProps.style, this.props.style )
+		);
+	}
+
+	clickItem = ( event: React.MouseEvent ) => {
```

Comment:
Definitely out of scope, but I wonder if we should be calling `event.preventDefault()` here. I've come to just do that subconsciously whenever writing event handlers like this. 

Hope not to be misleading. :-)

Reply:
I have to confess I didn't analyse the logic here deeply enough to make that call 😅 In general, I prefer to let events propagate as they normally would, unless there's a specific reason for not doing so. My (completely fallible and dubious) logic here would be that if had at some point been necessary, it would have been added already, so I'm reluctant to do so proactively.

Reply:
A `div` element, which the `clickItem` handler is attached to, doesn't have any default click behavior, so `preventDefault` is a noop that just confuses future generations 🙂 

It makes sense for cases where the `onClick` handler does a progressive enhancement of a static HTML.

Prevent submitting a HTML form and instead handle the submission by a script:
```jsx
<button type="submit" onClick={ handleSubmit } />
```

Prevent navigating to a link target and instead handle the navigation with a link:
```jsx
<a href={ url } onClick={ handleNavigation } />
```

Reply:
I switched this to a button, but I believe @jsnajdr's point remains, as it's not a submit type and thus shouldn't have any default behaviour.